main.cpp

```diff
@@ -419,6 +380,61 @@ IntType& IntType::divide( int rhs )
     return *this;
 }
 
+struct Point
+{
+    Point(float x_, float y_);
+    Point( const FloatType& x_, const FloatType& y_ );
+    Point( const DoubleType& x_, const DoubleType& y_ );
+    Point( const IntType& x_, const IntType& y_ );
+
+    Point& multiply(float m)
+    {
+        x *= m;
+        y *= m;
+        return *this;
+    }
+
+    Point& multiply(const FloatType& m);
+    Point& multiply(const DoubleType& m);
+    Point& multiply(const IntType& m);
+
+    void toString();
+
+private:
+    float x{0}, y{0};
+};
+
+Point::Point(float x_, float y_): x(x_), y(y_) { }
+
+Point::Point(const FloatType& x_, const FloatType& y_): Point(static_cast<float>(x_),static_cast<float>(y_)) { }
+
+Point::Point(const DoubleType& x_, const DoubleType& y_): Point(static_cast<float>(x_),static_cast<float>(y_)) { }
+
+Point::Point(const IntType& x_, const IntType& y_): Point(static_cast<float>(x_),static_cast<float>(y_)) { }
+
+void Point::toString()
+{
+    std::cout << "Point { x: " << x << ", y: " << y << " }\n";
+}
+
+//overloaded functions
+FloatType& FloatType::pow(const FloatType& f) { powInternal(f); return *this; }
+FloatType& FloatType::pow(const DoubleType& f) { powInternal(static_cast<float>(f)); return *this; }
+FloatType& FloatType::pow(const IntType& f) { powInternal(static_cast<float>(f)); return *this; }
+
+DoubleType& DoubleType::pow(const FloatType& d) { powInternal(static_cast<double>(d)); return *this; }
+DoubleType& DoubleType::pow(const DoubleType& d) { powInternal(d); return *this; }
+DoubleType& DoubleType::pow(const IntType& d) { powInternal(static_cast<double>(d)); return *this; }
+
+IntType& IntType::pow(const FloatType& i) { powInternal(static_cast<int>(i)); return *this; }
+IntType& IntType::pow(const DoubleType& i) { powInternal(static_cast<int>(i)); return *this; }
+IntType& IntType::pow(const IntType& i) { powInternal(i); return *this; }
+
+Point& Point::multiply(const FloatType& m) { x *= m; y *= m; return *this; }
+Point& Point::multiply(const DoubleType& m) { x *= static_cast<float>(m); y *= static_cast<float>(m);  return *this; }
+Point& Point::multiply(const IntType& m) { x *= static_cast<float>(m); y *= static_cast<float>(m);  return *this; }
```

Comment:
D.R.Y.
Call the `multiply(float)` function directly.  Return what it returns. 
- [ ] revise